hail/src/main/scala/is/hail/expr/ir/functions/LocusFunctions.scala

```diff
@@ -151,7 +151,7 @@ object LocusFunctions extends RegistryFunctions {
     val locusClass = Locus.getClass
 
     registerCode("contig", tlocus("T"), TString(),
-      (x: PType) => -x.asInstanceOf[PLocus].representation.asInstanceOf[PStruct].field("contig").typ) {
+      (x: PType) => -x.asInstanceOf[PLocus].representation.field("contig").typ) {
```

Comment:
I'm not sure if this and above constitute implementation changes. By putting representation on PLocus, there becomes no need for a cast.

Reply:
Notably, I need to have representation on PLocus, and I set that to `val representation: PStruct` to satisfy existing interfaces. I can changes this to `val representation: PType` as well.

Reply:
yeah, sounds good

Reply:
the right solution is probably to expose a `contigType: PString` and `positionType: PInt32` on PLocus.

Reply:
(in a followup)

Reply:
https://github.com/hail-is/hail/issues/7741